### DISCUSSION TITLE
Custom expiration time for custom auth token (within 1 hour limit by Firebase)

### DISCUSSION THREAD

**Discussion by cosmini** [kreait/firebase-php#653] [Q&A] [answered]
I was wondering if it possible to create a custom token with a custom expiration for example just 10 minutes instead of default 1 hour.

In `Kreait\Firebase\Auth::createCustomToken` we can provide custom claims, but there's no parameter for expiration.
```
public function createCustomToken($uid, array $claims = []): Token
```
In `Kreait\Firebase\Auth\CustomTokenViaGoogleIam::createCustomToken` however, this is accepted as 3rd param
```
public function createCustomToken($uid, array $claims = [], ?\DateTimeInterface $expiresAt = null): Token
```

Can someone help explain if it possible and if not, why this is not an option?

Thanks 👍

**Comment by jeromegamez** [comment, marked as answer]
This is now possible, with later versions of the SDK! 🙏
